event-details/LeaveEvent.test.tsx

```diff
@@ -0,0 +1,197 @@
+          event: { userAttendeeStatus: "attending" }
+        })
+      })
+      testEnv.leaveEvent.mockResolvedValueOnce("co-host-not-found")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await waitFor(() => {
+        expect(eventDetailsResult.current).toMatchObject({
+          event: { userAttendeeStatus: "hosting" }
+        })
+      })
+    })
+    it("should not call onSuccess on non-successful Leave Event result", async () => {
+      const { result: leaveEventResult } = renderUseLeaveEvent({
+        userAttendeeStatus: "attending"
+      })
+      testEnv.leaveEvent.mockResolvedValueOnce("co-host-not-found")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await verifyNeverOccurs(() =>
+        expect(testEnv.onSuccess).toHaveBeenCalled()
+      )
+    })
+    it("should change attendee status to not-participating when Leave Event returns success", async () => {
+      const { result: leaveEventResult } = renderUseLeaveEvent({
+        userAttendeeStatus: "attending"
+      })
+      const { result: eventDetailsResult } = renderUseEventDetails({
+        ...TEST_EVENT,
+        userAttendeeStatus: "attending"
+      })
+      await waitFor(() => {
+        expect(eventDetailsResult.current).toMatchObject({
+          event: { userAttendeeStatus: "attending" }
+        })
+      })
+      testEnv.leaveEvent.mockResolvedValueOnce("success")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await waitFor(() => {
+        expect(eventDetailsResult.current).toMatchObject({
+          event: { userAttendeeStatus: "not-participating" }
+        })
+      })
+    })
+    test("Error alerts appear correctly for error cases", async () => {
+      const { result: leaveEventResult } = renderUseLeaveEvent({
+        userAttendeeStatus: "attending"
+      })
+      testEnv.leaveEvent.mockResolvedValueOnce("event-has-ended")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await waitFor(() => {
+        expect(alertPresentationSpy).toHaveBeenCalledWith(
+          "Event has ended",
+          "This event has ended. You will be moved to the previous screen."
+        )
+      })
+
+      testEnv.leaveEvent.mockResolvedValueOnce("event-was-cancelled")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await waitFor(() => {
+        expect(alertPresentationSpy).toHaveBeenCalledWith(
+          "Event was cancelled",
+          "This event was cancelled. You will be moved to the previous screen."
+        )
+      })
+
+      testEnv.leaveEvent.mockResolvedValueOnce("co-host-not-found")
+      act(() => (leaveEventResult.current as any).confirmButtonTapped())
+      await waitFor(() => {
+        expect(alertPresentationSpy).toHaveBeenCalledWith(
+          "Event has no co-host",
+          "This event has no co-host. To leave, you will need to select a new host."
+        )
+      })
+    })
```

Comment:
We could just use the `LEAVE_EVENT_ERROR_ALERTS` object for these assertions instead of copy-pasting the messages.